rust/cubesql/cubesql/src/compile/engine/information_schema/postgres/pg_collation.rs

```diff
@@ -0,0 +1,259 @@
+            Arc::new(self.collname.finish()),
+            Arc::new(self.collnamespace.finish()),
+            Arc::new(self.collowner.finish()),
+            Arc::new(self.collprovider.finish()),
+            Arc::new(self.collisdeterministic.finish()),
+            Arc::new(self.collencoding.finish()),
+            Arc::new(self.colliculocale.finish()),
+            Arc::new(self.collcollate.finish()),
+            Arc::new(self.collctype.finish()),
+            Arc::new(self.colllocale.finish()),
+            Arc::new(self.collicurules.finish()),
+            Arc::new(self.collversion.finish()),
+        ];
+        columns
+    }
+}
+
+pub struct PgCatalogCollationProvider {
+    data: Arc<Vec<ArrayRef>>,
+}
+
+impl PgCatalogCollationProvider {
+    pub fn new() -> Self {
+        // See https://github.com/postgres/postgres/blob/REL_16_4/src/include/catalog/pg_collation.h
+        let mut builder = PgCatalogCollationBuilder::new();
+
+        // Initial contents of the pg_collation system catalog.
+        // See https://github.com/postgres/postgres/blob/REL_16_4/src/include/catalog/pg_collation.dat
+        builder.add_collation(&PgCollation {
+            oid: 100,
+            collname: "default",
+            collnamespace: PG_NAMESPACE_CATALOG_OID,
+            collowner: 10,
+            collprovider: "d".to_string(),
+            collisdeterministic: true,
+            collencoding: -1,
+            collcollate: None,
+            colliculocale: None,
+            collctype: None,
+            colllocale: None,
+            collicurules: None,
+            collversion: None,
+        });
+        builder.add_collation(&PgCollation {
+            oid: 950,
+            collname: "C",
+            collnamespace: PG_NAMESPACE_CATALOG_OID,
+            collowner: 10,
+            collprovider: "c".to_string(),
+            collisdeterministic: true,
+            collencoding: -1,
+            collcollate: Some("C".to_string()),
+            colliculocale: Some("C".to_string()),
+            collctype: Some("C".to_string()),
+            colllocale: None,
+            collicurules: None,
+            collversion: None,
+        });
+        builder.add_collation(&PgCollation {
+            oid: 951,
+            collname: "POSIX",
+            collnamespace: PG_NAMESPACE_CATALOG_OID,
+            collowner: 10,
+            collprovider: "c".to_string(),
+            collisdeterministic: true,
+            collencoding: -1,
+            collcollate: Some("POSIX".to_string()),
+            colliculocale: Some("POSIX".to_string()),
+            collctype: Some("POSIX".to_string()),
+            colllocale: None,
+            collicurules: None,
+            collversion: None,
+        });
+        builder.add_collation(&PgCollation {
+            oid: 962,
+            collname: "usc_basic",
+            collnamespace: PG_NAMESPACE_CATALOG_OID,
+            collowner: 10,
+            collprovider: "c".to_string(),
```

Comment:
Verified to be `b` on 17.6 in official Docker images.

```suggestion
            collprovider: "b".to_string(),
```